attestation-agent/kbs_protocol/src/client/rcar_client.rs

```diff
@@ -133,8 +134,13 @@ impl KbsClient<Box<dyn EvidenceProvider>> {
 
         debug!("get challenge: {challenge:#?}");
         let tee_pubkey = self.tee_key.export_pubkey()?;
-        let materials = vec![tee_pubkey.k_mod.as_bytes(), tee_pubkey.k_exp.as_bytes()];
-        let evidence = self.generate_evidence(challenge.nonce, materials).await?;
+        let runtime_data = json!({
+            "tee-pubkey": tee_pubkey,
+            "nonce": challenge.nonce,
+        });
```

Comment:
is there a reason why to combine them? My expectation was to have it the same way it's implemented in #369

Reply:
no matter Nonce or tee pub key, they all are things whose integrity should be protected by binding the evidence. There must be more scenarios except for the two fields.

We use a json map here to have a standard way to organize them

Reply:
this contributes to the hashing problem we've talked about. it's not only what hash algorithm is used but also what data is hashed. this won't work well with Intel Trust Authority which makes me think that perhaps we are going to need AS specific cfg flags to decide how the reportdata needs to be generated by the attesters

Reply:
see the runtime data spec https://github.com/confidential-containers/kbs/issues/259

This would be a good start to make runtime data usage more standard

Reply:
thanks! I saw it earlier but forgot to take a closer look. I'll do it today.